pkg/controller/snapshot_controller_base.go

```diff
@@ -238,10 +238,7 @@ func (ctrl *csiSnapshotController) snapshotWorker() {
 			glog.Errorf("expected vs, got %+v", vsObj)
 			return false
 		}
-		newSnapshot, err := ctrl.checkAndUpdateSnapshotClass(snapshot)
```

Comment:
This check is still needed because some CSI plugins need to use secrets stored in SnapshotClass to delete the snapshot.

Reply:
Then deletion will fail with some error msg, right?
The reason I don't want to put the check here is that if SnapshotClass is deleted, the snapshot status become not ready and user cannot use the snapshot to create volume.